plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -391,9 +1328,72 @@ func (c *CiscoTelemetryMDT) parseKeyField(tags map[string]string, field *telemet
 	}
 }
 
+func (c *CiscoTelemetryMDT) parseRib(grouper *metric.SeriesGrouper, field *telemetry.TelemetryField, prefix string, path string, tags map[string]string, timestamp time.Time) {
+	// RIB
+	measurement := path
+	for _, subfield := range field.Fields {
+		//For Every table fill the keys which are vrfName, address and masklen
+		switch subfield.Name {
+		case "vrfName", "address", "maskLen":
+			tags[subfield.Name] = decodeTag(subfield)
+		}
+		if value := decodeValue(subfield); value != nil {
+			grouper.Add(measurement, tags, timestamp, subfield.Name, value)
+		}
+		if subfield.Name == "nextHop" {
+			//For next hop table fill the keys in the tag - which is address and vrfname
+			for _, subf := range subfield.Fields {
+				for _, ff := range subf.Fields {
+					switch ff.Name {
+					case "address", "vrfName":
+						key := "nextHop/" + ff.Name
+						tags[key] = decodeTag(ff)
+					}
+					if value := decodeValue(ff); value != nil {
+						name := "nextHop/" + ff.Name
+						grouper.Add(measurement, tags, timestamp, name, value)
+					}
```

Comment:
Do you mean
```suggestion
					case "address", "vrfName":
						key := "nextHop/" + ff.Name
						tags[key] = decodeTag(ff)
						continue
					}
					if value := decodeValue(ff); value != nil {
						name := "nextHop/" + ff.Name
						grouper.Add(measurement, tags, timestamp, name, value)
					}
```

as otherwise you process `ff` twice in case of `address` and `vrfName`...